tests/generation/test_utils.py

```diff
@@ -1931,22 +1931,29 @@ def test_generate_with_quant_cache(self):
     @pytest.mark.generate
     @require_torch_gpu
     @slow
-    @is_flaky()  # compilation may result in equivalent (!= same) FP ops, causing the argmax in `generate` to be flaky
-    def test_generate_compile_fullgraph(self):
+    @parameterized.expand(
+        [
+            ("forward_only", False),  # TODO (@joao): a few models failing. After fixed, this should not be "@slow"
+            ("end_to_end", True),  # TODO (@joao): end-to-end compilation is broken with torch 2.5+, explore and fix
+        ]
+    )
+    def test_generate_compile(self, name, end_to_end):
```

Comment:
Instead of having a test for `torch.compile` against large checkpoints, let's do it against dummy checkpoints -- the importance of the test is to confirm a) we can compile AND b) compiled is equivalent to non-compiled.

Testing `model.forward` compilation is very similar to testing `model.generate` compilation, so `parameterized` is used. A few modifications were added to make the test more efficient (e.g. compile only once)

Note: this test doesn't have to be slow, but we currently have a few failures. Let's fix them in a follow-up PR